### ISSUE TITLE
What is the correct way of passing config.json file to the api

### ISSUE THREAD

**Issue by derick222** [jaggedsoft/php-binance-api#184]
Hi, I am trying to pass my config.json to the api

$api = new Binance\API("config.json");

but I keep getting this error:
Unable to load config from: C:\Users\r/.config/jaggedsoft/php-binance-api.json
Detected no API KEY or SECRET, all signed requests will fail
Unable to load config from: C:\Users\r/.config/jaggedsoft/php-binance-api.json
No proxies will be used
Unable to load config from: C:\Users\r/.config/jaggedsoft/php-binance-api.json
No curl options will be set

In php-binance-api.php line 858:

  signedRequest error: API Key not set!

I tried to check my config.json path and this the return:

$file = C:\wamp64\www\dev-jaggedsoft\config\jaggedsoft\config.json

and if I tried to put manually the file name like this , It works
 case 0:
                $this->setupApiConfigFromFile("config.json");
                $this->setupProxyConfigFromFile();
                $this->setupCurlOptsFromFile();
                break;
and changing this line of code:
$file = is_null($file) ? getenv("HOME") . "/.config/jaggedsoft/php-binance-api.json" : "C:\wamp64\www\dev-jaggedsoft\config\jaggedsoft\\".$file;
        
Any help is appreciated!
Thanks

**Comment by jaggedsoft** [comment]
Yeah PHP might require the path to be like this since you're on windows:
`$api = new Binance\API("file://wamp64/www/dev-jaggedsoft/config/jaggedsoft/config.json");`
Good luck!

**Comment by dmzoneill** [comment]
also you can use getcwd to reolve the path

```
 new Binance\API( getcwd() . "/config.json" );
```

**Comment by derick222** [comment]
Hi, because I don't want to display the complete path in the code, so instead, I used this in my code

my actual code
$api = new Binance\API($input->getArgument('APIKeys'));

and the $input->getArgument('APIKeys') is equal to config.json filename

What I want is to just pass the config.json file, or in the future it will be like client01.json, client02.json etc...

Thanks for a very fast reply

**Comment by dmzoneill** [comment]
```
$api = new Binance\API( getcwd() .  "/" . $input->getArgument('APIKeys'));
```

**Comment by derick222** [comment]
Now it sees the file path, but still having an error.
Unable to load config from: C:\wamp64\www\dev-jaggedsoft/config.json
Detected no API KEY or SECRET, all signed requests will fail
Unable to load config from: C:\wamp64\www\dev-jaggedsoft/config.json
No proxies will be used
Unable to load config from: C:\wamp64\www\dev-jaggedsoft/config.json
No curl options will be set
Unable to load config from: C:\Users\respina/config/jaggedsoft/php-binance-api.json
Detected no API KEY or SECRET, all signed requests will fail
Unable to load config from: C:\Users\respina/config/jaggedsoft/php-binance-api.json
No proxies will be used
Unable to load config from: C:\Users\respina/config/jaggedsoft/php-binance-api.json
No curl options will be set

**Comment by dmzoneill** [comment]
it cant find your file.  you will need to provide a full path.
if you are not executing the file within the same directory as the files

WORKDIR is not always the same as basenase( __FILE__ ).

for eg. if you have the file **test.php**

if you are in **c:\scripts**
php test.php
new $api = new Binance\API( $input->getArgument('APIKeys'));
// equivalent to c:\directroy\config.json

if you are in c:\
php scripts\test.php
// equivalent to c:\config.json


for your purposes you cant use a relative path, unless you have the files in WORKDIR and are relative.

**Comment by derick222** [comment]
Hi, I have tried everything for the path in ubuntu server, but I still have problem understanding this line of code:
$file = is_null($file) ? getenv("HOME") . "/.config/jaggedsoft/php-binance-api.json" : $file;

I have passed a file parameter to the api, but the code above sees that $file is null, which is not
$api = new Binance\API( getcwd() .  "/" . $input->getArgument('APIKeys'));

**Comment by dmzoneill** [comment]
Its called php urinary operator
```
$something = (expression) ? True value; false value;
```
Equivalent to
```
$something = null;
If( expression )
   $something = true;
else
   $something = false;
```

**Comment by derick222** [comment]
I tried to check if the path that I was passing is correct or not by changing the default path and the code works, It means that the path that I am passing is correct. 
$file = is_null($file) ? getcwd() . "/config/datas/cr.json" : $file;

**Comment by dmzoneill** [comment]
Once again. There is no reason to modify the code. Is_null is failing for you.. meaning you are most likely passing in a null variable

**Comment by derick222** [comment]
it return:
bool(true)

**Comment by dmzoneill** [comment]
What happens if you put this file in
~/.config/jaggedsoft/php-binance-api.json

```
new Binance\Api(); //no parameters
```

**Comment by dmzoneill** [comment]
What happens if you put this file in
~/.config/jaggedsoft/php-binance-api.json

```
new Binance\Api(); //no parameters
```

**Comment by derick222** [comment]
the parameter works, I created this 
~/.config/jaggedsoft/php-binance-api.json
and put my keys in the json file.  the program works.

**Comment by dmzoneill** [comment]
Apologies for the delayed response.
I'm sorry to inform you that the error is on your side (incorrect path, unreadable file,....), consider as of just now.

```
git clone https://github.com/jaggedsoft/php-binance-api.git

cd in php-binance-api/examples/

cat >  test.json << EOF
{
    "api-key": "5enQYcMQk2J3syHCao9xgJOnnPoGtDMhSRRAzG2Gxo90TBzXPG1itcXikQc2VRDh",
    "api-secret": "uWJQXigS3AjftKe8c6xK2t3rkTqkmfeeNPwcycBLGXXsuU4eUvLkPY9qcOnB2UYI"
}
EOF

cat >  test.php << EOF
<?php
require '../php-binance-api.php';
\$api = new Binance\API("test.json");
print_r( \$api->Balances() );
EOF

php test.php 
Array
(
    [BTC] => Array
        (
            [available] => 0.00000000
            [onOrder] => 0.00000000
            [btcValue] => 0
            [btcTotal] => 0
        )
.....
```

**Comment by derick222** [comment]
What is your complete path for the test.json file?

**Comment by derick222** [comment]
Taking new approach.

**Comment by hetesk** [comment]
None of the possible solutions above works.
